src/views/tutorial-library/utils/scroll.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef, useEffect } from 'react'
+import { useInstantSearch } from 'react-instantsearch-hooks-web'
+
+/**
+ * Scrolls to the top of the page when results have meaningfully changed from user action.
+ * We do this by ignoring the query for initial results, and only scrolling ony subsequent searches.
+ */
+export const useScrollToTopOnResultsChange = () => {
```

Comment:
I don't think we should auto-scroll the page when results change. I'm not sure if this technically fails [Success Criterion 2.2.2: Pause, Stop, Hide (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html), but the intent behind the SC is still relevant here:

- Content that moves or auto-updates can be a barrier to anyone who has:
  - trouble reading stationary text quickly
  - trouble tracking moving objects
- It can also cause problems for screen readers
- Moving content can also be a severe distraction for some people

Reply:
This is a tricky one 💭 I think of applying filters as a form of navigation in this case, especially so given that the URL is updating along with the results changing. Thanks for linking the success criterion! Reading through it, it seems really focused on auto-updating / motion / live content.

> "Moving, blinking and scrolling" refers to content in which the visible content conveys a sense of motion. Common examples include motion pictures, synchronized media presentations, animations, real-time games, and scrolling stock tickers.

Looking at this as a form of navigation, the page scrolling to the top doesn't feel unexpected, and I would actually expect it to happen.

With that being said, I'm happy to remove it based on the spirit of 2.2.2, but I wanted to share some additional context for why it was added!

Reply:
Definitely understand the context! I do think there will be some manual focus management needed when pagination is interacted with, and some scrolling will be a natural side effect of that. I think where the auto scrolling is less expected is when the sidebar filters are selected and something is typed in the search input, since those aren't explicitly conveyed as interactions that change the page URL. I do think it would be nice if we could remove for now, and then create a research ticket that we can come back to for refinement! 💖

Reply:
Sure thing!